packages/ts/src/components/axis/index.ts

```diff
@@ -362,33 +481,86 @@ export class Axis<Datum> extends XYComponentCore<Datum, AxisConfigInterface<Datu
   }
 
   private _renderAxisLabel (selection = this.axisGroup): void {
-    const { type, label, labelMargin, labelFontSize } = this.config
+    const { type, label, labelMargin, labelFontSize, labelTextFitMode } = this.config
 
     // Remove the old label first to calculate the axis size properly
     selection.selectAll(`.${s.label}`).remove()
 
+    if (!label) return
+
     // Calculate label position and rotation
     const axisPosition = this.getPosition()
     // We always use this.axisRenderHelperGroup to calculate the size of the axis because
     //    this.axisGroup will give us incorrect values due to animation
     const { width: axisWidth, height: axisHeight } = this._axisRawBBox ?? selection.node().getBBox()
 
-    const offsetX = type === AxisType.X ? this._width / 2 : (-1) ** (+(axisPosition === Position.Left)) * axisWidth
-    const offsetY = type === AxisType.X ? (-1) ** (+(axisPosition === Position.Top)) * axisHeight : this._height / 2
-
-    const marginX = type === AxisType.X ? 0 : (-1) ** (+(axisPosition === Position.Left)) * labelMargin
-    const marginY = type === AxisType.X ? (-1) ** (+(axisPosition === Position.Top)) * labelMargin : 0
-
     const rotation = type === AxisType.Y ? -90 : 0
-    // Append new label
-    selection
+
+    // Create the text element (without transform first)
+    const textElement = selection
       .append('text')
       .attr('class', s.label)
-      .text(label)
       .attr('dy', `${this._getLabelDY()}em`)
-      .attr('transform', `translate(${offsetX + marginX},${offsetY + marginY}) rotate(${rotation})`)
       .style('font-size', labelFontSize)
       .style('fill', this.config.labelColor)
+
+    // Set the text content
+    textElement.text(label)
+
+    // Apply wrapping if enabled
+    if (labelTextFitMode === FitMode.Wrap) {
+      if (type === AxisType.Y) {
+        // For Y-axis, use the chart height as the maximum width before rotation
+        const maxWidthBeforeRotation = this._height
+
+        wrapSVGText(textElement, maxWidthBeforeRotation)
+      } else {
+        // For X-axis, use the chart width as the maximum width
+        const maxWidth = this._width
+        wrapSVGText(textElement, maxWidth)
+      }
+    }
+
+    // Calculate offset after wrapping to get accurate dimensions
+    let labelWidth = axisWidth
+    let labelHeight = axisHeight
+    if (labelTextFitMode === FitMode.Wrap) {
+      const labelBBox = textElement.node().getBBox()
+      labelWidth = labelBBox.width
+      labelHeight = labelBBox.height
+    } else {
+      const trimWidth = type === AxisType.X ? labelWidth : labelHeight
+      trimSVGText(
+        textElement,
+        trimWidth,
+        this.config.labelTextTrimType as TrimMode,
+        true,
+        parseInt(labelFontSize || '12', 10),
+        0.58
```

Comment:
I don't think we should hardcode the width to hight ratio. Users might use different fonts